app/src/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2857,7 +2920,11 @@ public boolean onFling(MotionEvent e1, MotionEvent e2, float velocityX, float ve
             if (velocityX > 0) {
                 showPreviousView();
             } else {
-                showNextView();
+                int event = mFormController.getEvent(mFormController.getNextFormIndex(mFormController.getFormIndex(), true));
+                boolean navBar = PreferencesActivity.getProgressBarMode(this).useNavigationBar();
```

Comment:
I like to give boolean variables names that sound like questions, like 'hasNavBar' or 'usingNavBar'. This helps (whatever small amount) when reading complex boolean expressions.